packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -135,9 +154,16 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     final BuildableIOSApp app = await buildableIOSApp;
     Status? status;
     RunResult? result;
-    final String outputPath = globals.fs.path.absolute(app.ipaOutputPath);
+    final String relativeOutputPath = app.ipaOutputPath;
+    final String absoluteOutputPath = globals.fs.path.absolute(relativeOutputPath);
+    final String archivePath = globals.fs.path.absolute(app.archiveBundleOutputPath);
+    final String? exportMethod = stringArg('export-method');
+    final bool isAppStoreUpload = exportMethod  == 'app-store';
     try {
-      status = globals.logger.startProgress('Building IPA...');
+      final String? ipaType = isAppStoreUpload ? 'App Store' : exportMethod;
```

Comment:
If `exportMethod` can be `null` here you might do:
```suggestion
      final String? ipaType = isAppStoreUpload ? 'App Store' : (exportMethod ?? 'something');
```

Reply:
Why is `app-store` a special case here? is `ipaType` only used in the print?

Reply:
> If `exportMethod` can be `null` here you might do:

Actually it can't be null, it's an option with `allowed` and `defaultsTo`.  I'll make this a `!`.

> Why is `app-store` a special case here? is `ipaType` only used in the print?

It's just for display in the logging, `Build app-store IPA` doesn't make sense unless you already know the guts of the keys of the plist, compared to `Build App Store IPA`.
I'll rename the variable.